o1vm/src/mips/constraints.rs

```diff
@@ -451,58 +470,45 @@ impl<Fp: Field> InterpreterEnv for Env<Fp> {
                 );
             }
 
-            // Constrain booleanity of has_n_bytes (at least the one for 1 byte must be 1)
-            // TODO: could it read 0 bytes?
-            {
-                self.constraints
-                    .push(reading_preimage.clone() * (has_n_bytes[0].clone() - Expr::from(1)));
-                for flag in &has_n_bytes[1..] {
-                    self.constraints.push(
-                        reading_preimage.clone() * flag.clone() * (flag.clone() - Expr::from(1)),
-                    );
-                }
-            }
-
-            // Constrain the bytes flags depending on the number of bytes read in this row
+            // Constrain the bytes flags depending on the number of bytes read
+            // in this row
             {
-                // When at least has_1_byte, then any number of bytes can be read
-                // <=> Check that you can only read 1, 2, 3 or 4 bytes
+                // When at least has_1_byte, then any number of bytes can be
```

Comment:
Same here. Only important change is `read_preimage` has been removed (expected).